docs/reference/deployment-targets-yaml-reference.md

```diff
@@ -0,0 +1,271 @@
+
+### metadata
+
+The `metadata` property defines the metadata for a Deployment Target YAML file,
+supporting the following properties: `id`, `name`, `project_id`, `organization_id`,
+`created_by`, `created_at`, `updated_by`, `updated_at`, `description` and `url`
+properties.
+
+#### id
+
+The `id` property is an automatically generated unique identifier assigned by
+Semaphore 2.0 to each deployment target. It should not be modified.
+
+#### name
+
+The `name` property is a string that represents the name of the deployment target.
+This name is displayed when executing the `sem get dt -p [project-name]` command.
+The name should only consist of alphanumerical characters, dashes, underscores, and dots.
+
+#### project_id
+
+The `project_id` property corresponds to the UUID (Universally Unique Identifier) of
+the project associated with the deployment target. It should not be modified.
+
+#### organization_id
+
+The `organization_id` property represents the UUID of the organization linked to
+the deployment target. It should not be modified.
+
+#### created_by
+
+The `created_by` property contains the UUID of the user who created the deployment
+target. It should not be modified.
+
+#### created_at
+
+The `created_at` property indicates the UTC when the deployment target was created.
+It should not be modified.
+
+#### updated_by
+
+The `updated_by` property stores the UUID of the user who last updated the deployment
+target. It should not be modified.
+
+#### updated_at
+
+The `updated_at` property denotes the UTC when the deployment target was last
+updated. It should not be modified.
+
+#### description
+
+The `description` property is a string that describes the deployment target.
+
+#### url
+
+The `url` property is a string containing the URL associated with the deployment target.
+
+### spec
+
+The `spec` property contains a list of deployment target properties.
+
+#### state
+
+The `state` property indicates the state of the deployment target, which can be
+one of the following values: `SYNCING`, `USABLE`, `UNUSABLE`, or `CORDONED`. It should
+not be modified.
+
+#### state_message
+
+The `state_message` property provides an additional message associated with
+the deployment target's state. It should not be modified.
+
+#### subject_rules
+
+The `subject_rules` property holds a list of subject rules that determine who
+can trigger promotions when the deployment target is associated with them. Each
+rule must include the `type`. Depending on the rule `type`, the `subject_id`
+or `git_login` may also be required.
+
+##### type
+
+The `type` property of a subject rule can have the following values:
+`ANY`, `USER`, `ROLE`, `GROUP`, or `AUTO`. If `ANY` is used, any user can trigger
+the promotion. If `USER` is used, only specified users (who must be assigned to
+the project) can trigger a promotion. For `ROLE`, only specified user roles defined
+within the project can trigger a promotion.
```

Comment:
We should also explain what `AUTO`  is here. Also, remove `GROUP` for now since we do not support them yet.